File: Backend/app/Models/Product.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Facades\Storage;

class Product extends Model
{
    /** @use HasFactory<\Database\Factories\ProductFactory> */
    use HasFactory;

    protected $fillable = [
        'name',
        'description',
        'category_id',
        'total_rating',
        'price',
        'stock_quantity',
        'total_review_count',
        'vendor_id',
        'image_path'
    ];

    public function category()
    {
        return $this->belongsTo(Category::class);
    }

    public function vendor()
    {
        return $this->belongsTo(Vendor::class);
    }

    public function getImageUrl()
    {
        return $this->image_path
            ? asset('storage/' . $this->image_path)
            : null;
    }
}

File: Flutter/lib/data/models/product.dart

class Product {
  int? id;
  String? name;
  String? description;
  double? price;
  String? categoryName;
  int? categoryId;
  double? rating;
  int? stockQuantity;
  int? totalReviewCount;
  String? vendorName;
  int? vendorId;

  Product({this.id, this.name, this.description, this.price, this.categoryName, this.categoryId, this.rating, this.stockQuantity, this.totalReviewCount, this.vendorId, this.vendorName});

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      price: json['price'],
      categoryId: json['categoryId'],
      categoryName: json['categoryName'],
      rating: json['rating'],
      stockQuantity: json['stockQuantity'],
      totalReviewCount: json['totalReviewCount'],
      vendorId: json['vendorId'],
      vendorName: json['vendorName']
    );
  }
}
